src/ClapDetector.js

```diff
@@ -0,0 +1,168 @@
+                ]).then((analysers) => {
+                    const defaultInputAnalyser = analysers[0];
+                    const externalInputAnalyser = analysers[1];
+                    const wirelessInputAnalyser = analysers[2];
+                    const scriptProcessorAnalysisNode = context.createScriptProcessor(2048,2,1);
+                    const audioGain = context.createGain();
+                    scriptProcessorAnalysisNode.connect(audioGain);
+                    audioGain.connect(context.destination);
+                    scriptProcessorAnalysisNode.onaudioprocess = () => {
+
+                        let sim1 = this.getClapSimilarity(defaultInputAnalyser, 5, 1);
+                        let sim2 = this.getClapSimilarity(externalInputAnalyser, 5, 2);
+                        let sim3 = this.getClapSimilarity(wirelessInputAnalyser, 5, 3);
+
+                        if (sim1 >= sim2 && sim1 >= sim3 && sim1 >= 0.90) {
+                            console.log(sim1);
+                            console.log('left');
+                            this.props.onDetect('left');
+                        } else if (sim2 >= sim1 && sim2 >= sim3 && sim2 >= 0.90) {
+                            console.log(sim2);
+                            console.log('right');
+                            this.props.onDetect('right');
+                        } else if (sim3 >= sim1 && sim3 >= sim2 && sim3 >= 0.90) {
+                            console.log(sim3);
+                            console.log('forward');
+                            this.props.onDetect('forward');
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    addAudioStreamNode = async (deviceId: string, audioContext: any): any=> {
+        const audioInputDevice = await navigator.mediaDevices.getUserMedia({audio: {deviceId: this.state.audioInputDeviceIds[0]}});
+        const audioInputStream = audioContext.createMediaStreamSource(audioInputDevice);
+        const audioAnalyser = audioContext.createAnalyser();
+        audioAnalyser.smoothingTimeConstant = 0;
+        audioAnalyser.fftSize = 2048;
+        audioInputStream.connect(audioAnalyser);
+        const bufferLength = audioAnalyser.frequencyBinCount;
+        const arrayFreqDomain = new Uint8Array(bufferLength);
+        const arrayTimeDomain = new Uint8Array(bufferLength);
+        return ({
+            analyser : audioAnalyser,
+            freqBuffer : arrayFreqDomain,
+            timeBuffer : arrayTimeDomain
+        });
+    }
+
+    getClapSimilarity(analyserNode: any, numRows: number, deviceId: number): number {
+        let timeDomainBuffer = analyserNode.analyser.getByteTimeDomainData(analyserNode.timeBuffer);
+        let freqDomainBuffer = analyserNode.analyser.getByteFrequencyData(analyserNode.freqBuffer);
+        let currentTimeValue = (analyserNode.timeBuffer[0] / 128) - 1.0;
```

Comment:
getByteTimeDomainData (and the newer getFloatTimeDomainData) return an array of the size you requested - its frequencyBinCount, which is calculated as half of the requested fftSize. That array is, of course, at the current sampleRate exposed on the AudioContext, so if it's the default 2048 fftSize, frequencyBinCount will be 1024, and if your device is running at 44.1kHz, that will equate to around 23ms of data.

The byte values do range between 0-255, and yes, that maps to -1 to +1, so 128 is zero. (It's not volts, but full-range unitless values.)

If you use getFloatFrequencyData, the values returned are in dB; if you use the Byte version, the values are mapped based on minDecibels/maxDecibels (see the minDecibels/maxDecibels description).

--https://stackoverflow.com/questions/24083349/understanding-getbytetimedomaindata-and-getbytefrequencydata-in-web-audio

Reply:
Since analyserNode.timeBuffer array contains values in range between 0 to 255 or -1 to +1 accordingly, by dividing the value from the buffer by 128 and -1 will return actual value; for example, 128 should be 0, so (128 / 128) - 1.0 = 0.